syntax = "proto3";
package srv6pm;
import "srv6pmCommons.proto";
import "commons.proto";

service SRv6PMReflector {
    // start an experiment
    rpc startExperiment (StartExperimentReflectorRequest) returns (StartExperimentReflectorReply) {}
    // stop am experiment
    rpc stopExperiment (StopExperimentRequest) returns (StopExperimentReply) {}
    // retrive data of an experiment
    rpc retriveExperimentResults (RetriveExperimentDataRequest) returns (ExperimentDataResponse) {}
    // Tunnel
    rpc CreateSRv6TunnelReflector (SRv6EPRequestReflector) returns (SRv6EPReplyReflector) {}

}

// The SRv6EPRequest message containing a number of paths.
message SRv6EPRequestReflector {
    string prefix= 1;
    string encapmode = 2;
    string segments = 3;
    string device = 4;
}

// The SRv6EPReply message containing a simple status message
message SRv6EPReplyReflector {
    srv6_services.StatusCode status = 200;
}

message StartExperimentReflectorRequest {
    uint32 measure_id = 1;
    string sdlist = 2;
    string sdlistreverse = 3;
    ReflectorOptions reflector_options = 4;
}

message StartExperimentReflectorReply {
    srv6_services.StatusCode status = 1;
}

message ReflectorOptions {
    //uint32 ss_udp_port = 1;
    //uint32 refl_udp_port = 2;
    MeasurementProtocol measurement_protocol = 3;
    AuthenticationMode authentication_mode = 4;
    MeasurementType measurement_type = 5;
    MeasurementLossMode measurement_loss_mode = 6;
    string authentication_key = 7;
}
